core/src/code.rs

```diff
@@ -169,25 +176,28 @@ fn check_imports(module: &Module) -> Result<(), CodeError> {
             .iter()
             .copied()
             .map(Into::<ValueType>::into);
-        if !params.eq(func_type.params().iter().copied()) {
-            return Err(CodeError::InvalidImportFnSignature);
-        }
-
         let results = signature.results().unwrap_or(&[]);
-        if results != func_type.results() {
-            return Err(CodeError::InvalidImportFnSignature);
+
+        if !(params.eq(func_type.params().iter().copied()) && results == func_type.results()) {
```

Comment:
```suggestion
        if !(params == func_type.params().iter().copied() && results == func_type.results()) {
```

Reply:
it doesn't compile: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=604199afd4e37120050b97ec0930d7ce